docs/api/js/types.md

```diff
@@ -1,226 +1,303 @@
-    logs: Array<Log>;
-    l2ToL1Logs: Array<L2ToL1Log>;
-}
-
-export interface Block extends providers.Block {
-    l1BatchNumber: number;
-    l1BatchTimestamp: number;
-}
-
-export interface BlockWithTransactions extends EthersBlockWithTransactions {
-    l1BatchNumber: number;
-    l1BatchTimestamp: number;
-    transactions: Array<TransactionResponse>;
-}
-
-export interface Log extends providers.Log {
-    l1BatchNumber: number;
-}
-
-export interface L2ToL1Log {
-    blockNumber: number;
-    blockHash: string;
-    l1BatchNumber: number;
-    transactionIndex: number;
-    shardId: number;
-    isService: boolean;
-    sender: string;
-    key: string;
-    value: string;
-    transactionHash: string;
-    logIndex: number;
-}
-
-export type TransactionRequest = providers.TransactionRequest & {
-    customData?: Eip712Meta;
-};
-
-export interface PriorityOpResponse extends TransactionResponse {
-    waitL1Commit(confirmation?: number): Promise<providers.TransactionReceipt>;
-}
-
-export type BalancesMap = { [key: string]: BigNumber };
-
-export interface DeploymentInfo {
-    sender: Address;
-    bytecodeHash: string;
-    deployedAddress: Address;
-}
-
-export interface ApprovalBasedPaymasterInput {
-    type: 'ApprovalBased';
-    token: Address;
-    minimalAllowance: BigNumber;
-    innerInput: BytesLike;
-}
-
-export interface GeneralPaymasterInput {
-    type: 'General';
-    innerInput: BytesLike;
-}
-
-export interface EthereumSignature {
-    v: number;
-    r: BytesLike;
-    s: BytesLike;
-}
-
-export type PaymasterInput = ApprovalBasedPaymasterInput | GeneralPaymasterInput;
-
-export enum AccountAbstractionVersion {
-    None = 0,
-    Version1 = 1
-}
-
-export enum AccountNonceOrdering {
-    Sequential = 0,
-    Arbitrary = 1
-}
-
-export interface ContractAccountInfo {
-    supportedAAVersion: AccountAbstractionVersion;
-    nonceOrdering: AccountNonceOrdering;
-}
-
-export interface BatchDetails {
-    number: number;
-    timestamp: number;
-    l1TxCount: number;
-    l2TxCount: number;
-    rootHash?: string;
-    status: string;
-    commitTxHash?: string;
-    committedAt?: Date;
-    proveTxHash?: string;
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-    l1GasPrice: number;
-    l2FairGasPrice: number;
-}
-
-export interface BlockDetails {
-    number: number;
-    timestamp: number;
-    l1BatchNumber: number;
-    l1TxCount: number;
-    l2TxCount: number;
-    rootHash?: string;
-    status: string;
-    commitTxHash?: string;
-    committedAt?: Date;
-    proveTxHash?: string;
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-}
-
-export interface TransactionDetails {
-    isL1Originated: boolean;
-    status: string;
-    fee: BigNumberish;
-    initiatorAddress: Address;
-    receivedAt: Date;
-    ethCommitTxHash?: string;
-    ethProveTxHash?: string;
-    ethExecuteTxHash?: string;
-}
-```
+# Types and interfaces
+
+## `AccountAbstractionVersion`
```

Comment:
nit: i would add more detailed  explelation what is the difference.
cc @StanislavBreadless

Reply:
I'm being precise, we called the doc Types previously but as I went through I realised they're not all types, they're interfaces and enums also .. I can't call an interface, or an enum a type

Reply:
Oh yes, there is one comment ... please check this in new branch.